src/mame/drivers/ms0515.cpp

```diff
@@ -115,9 +211,128 @@ WRITE16_MEMBER(ms0515_state::ms0515_bank_w)
 	}
 }
 
-WRITE8_MEMBER(ms0515_state::ms0515_sys_w)
+WRITE16_MEMBER(ms0515_state::ms0515_vram_w)
+{
+	m_video_ram[offset*2]   = data;
+	m_video_ram[offset*2+1] = data >> 8;
+}
+
+READ16_MEMBER(ms0515_state::ms0515_halt_r)
+{
+	DBG_LOG(2,"Halt",("== %06o\n", m_haltreg));
+
+	return m_haltreg;
+}
+
+WRITE16_MEMBER(ms0515_state::ms0515_halt_w)
 {
-	m_sysreg = data;
+	DBG_LOG(1,"Halt",("<- %06o\n", data));
+
+	COMBINE_DATA(&m_haltreg);
+}
+
+/*
+ * b7 -- ROM bank
+ * b6 -- cassette data out
+ * b5 -- LED VD16
+ * b4 -- LED VD9
+ * b3 -- floppy side select (?? 1 -- top)
+ * b2 -- floppy motor (0 -- on)
+ * b1-0 -- floppy drive select
+ *
+ * DZ0 = drive 0 side 0 (bottom)
+ * DZ1 = drive 1 side 0 (bottom)
+ * DZ2 = drive 0 side 1 (top)
+ * DZ3 = drive 1 side 1 (top)
+ *
+ * MZ1 = drive 1 side 0-1
+ */
+WRITE8_MEMBER(ms0515_state::ms0515_porta_w)
+{
+	DBG_LOG(2,"Sysreg",("A <- %02x\n", data));
+
+	output().set_value("led16", BIT(data, 5));
+	output().set_value("led9", BIT(data, 4));
+
+	switch (data & 3)
+	{
+	case 0:
+		m_floppy = m_floppy0;
+		break;
+
+	case 1:
+		m_floppy = m_floppy1;
+		break;
+
+	default:
+		m_floppy = nullptr;
+		break;
+	}
+
+	if (m_floppy) {
+		m_fdc->set_floppy(m_floppy);
+		m_floppy->ss_w(!BIT(data, 3));
+		m_floppy->mon_w(BIT(data, 2));
+	} else {
+		m_floppy0->mon_w(1);
+		m_floppy1->mon_w(1);
+	}
+
+	m_sysrega = data;
+}
+
+/*
+ * b7 -- cassette data in
+ * b6-5 -- reserved for IRPR-M (parallel) port
+ * b4-3 -- DIP switches on video board, 00 -- 50 Hz, 01 -- 60 Hz, 1x -- 72 Hz
+ * b2 -- floppy ready signal (0 -- ready)
+ * b1 -- floppy drq (1 -- ready)
+ * b0 -- floppy intrq (0 -- ready)
+ */
+READ8_MEMBER(ms0515_state::ms0515_portb_r)
+{
+	uint8_t data;
+
+	data  = m_fdc->intrq_r();
+	data |= m_fdc->drq_r() << 1;
+
+	if (m_floppy) {
```

Comment:
Here you're using K&R brace style for a single-statement body where before you've omitted braces entirely.